### DISCUSSION TITLE
BG Datatable Editor

### DISCUSSION THREAD

**Discussion by NobodyYouKnow** [NightscoutFoundation/xDrip#2157] [General]
I find myself in occasional need to delete / revise erroneous data captured by xDrip. Is there an editor available for this task? Thanks!

**Comment by Navid200** [comment]
What exactly do you mean by delete?
If you don't want a point to be taken into account for statistical analysis, you can use the data table and mark it as a bad reading.  Its color will change on screen and it will not be used for statistical analysis.

**Comment by NobodyYouKnow** [comment]
Thanks for that suggestion. I do know about the ability to cause xdrip to ignore data from the data table. That erroneous data is still exported out unchanged to external repositories (think services like TidePool here). This is why I am asking about an ability to correct / delete the data within xdrip, not just ignore it.

**Comment by Navid200** [comment]
Even if we add a feature to delete a datapoint, you will still need to log into your Nightscout database separately to delete the point from that database as well.

**Reply by NobodyYouKnow** [reply]
That is an acceptable extra step - thanks! I hate propagating contaminated data. How do I add this as a feature request to XDrip? I have never done this within GitHub. Is this done as an Issue?

**Reply by Navid200** [reply]
Only if there is a point.
If there is absolutely no way a feature is going to be developed, it's pointless to add a feature request for it.  
That's the purpose of the conversation here.

Can you explain why this is necessary.

Even if someone develops this and creates a PR for it, the head developer is going to ask "why do we need this".
If there is no answer, the PR will not be merged.  Then, what's the point?

So, let's say you have one erroneous point in Nightscout.  Why is that a problem?

**Reply by Navid200** [reply]
You are contradicting yourself.

>  That erroneous data is still exported out unchanged to external repositories

When I responded to that and stated that you will have to delete the wrong data from the external repository, you said:

> That is an acceptable extra step  

Then, the question that comes after that is why do we need to delete the erroneous data from xDrip if the problem is the external repository?

**Comment by NobodyYouKnow** [comment]
First, my apologies for having provoked you. That was not my intent.

Thank you for the advice that this data integrity tool does not currently exist within XDrip. That knowledge will inform my next actions.

Regards.
